src/writeData/components/PluginAddToExistingConfiguration/Configure.tsx

```diff
@@ -0,0 +1,143 @@
+// Libraries
+import React, {FC} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
+
+// Components
+import {ComponentStatus, Dropdown} from '@influxdata/clockface'
+
+// Types
+import {AppState, Bucket, ResourceType, Telegraf} from 'src/types'
+import {PluginConfigurationStepProps} from 'src/writeData/components/PluginAddToExistingConfiguration/Wizard'
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+import {getDataLoaders} from 'src/dataLoaders/selectors'
+
+// Actions
+import {setBucketInfo} from 'src/dataLoaders/actions/steps'
+import {
+  setTelegrafConfigDescription,
+  setTelegrafConfigID,
+  setTelegrafConfigName,
+} from 'src/dataLoaders/actions/dataLoaders'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = PluginConfigurationStepProps & ReduxProps
+
+const ConfigureComponent: FC<Props> = props => {
+  const {
+    buckets,
+    onSetBucketInfo,
+    onSetTelegrafConfigDescription,
+    onSetTelegrafConfigID,
+    onSetTelegrafConfigName,
+    setIsValidConfiguration,
+    setPluginConfig,
+    telegrafConfigID,
+    telegrafs,
+  } = props
+
+  const selectedTelegraf = telegrafs?.find(
+    telegraf => telegraf.id === telegrafConfigID
+  )
+
+  if (!selectedTelegraf) {
+    setIsValidConfiguration(false)
+  }
+
+  const selectedName = selectedTelegraf
+    ? selectedTelegraf.name
+    : 'Select a configuration'
+
+  const handleSelectTelegrafConfiguration = (telegraf: Telegraf) => {
```

Comment:
if `buckets` are passed into this function, it can be created outside of the function component so it's not re-created on every render. if you name it by doing something like

```ts
const handleSelectTelegrafConfiguration = function handleSelectTelegrafConfiguration() {
```
it should re-render even less. When react receives an anonymous function, it will always re-render it since it's different from the previous anonymous function, even if the bodies do the same thing.

Reply:
i made a long video of this and pinged you in Slack.

Reply:
tl:dw: naming these doesn't affect re-rendering in this component tree